playbooks/openstack/advanced-configuration.md

```diff
@@ -482,65 +493,40 @@ spec:
+    openshift_hosted_registry_storage_kind: openstack
+    openshift_hosted_registry_storage_access_modes: ['ReadWriteOnce']
+    openshift_hosted_registry_storage_openstack_filesystem: ext4
+    openshift_hosted_registry_storage_volume_name: registry
+    openshift_hosted_registry_storage_volume_size: 10Gi
 
-This will use the credentials from your shell environment. If you want
-to enter them explicitly, you can. You can also use credentials
-different from the provisioning ones (say for quota or access control
-reasons).
 
-**NOTE**: If you're testing this on (DevStack)[devstack], you must
-explicitly set your Keystone API version to v2 (e.g.
-`OS_AUTH_URL=http://10.34.37.47/identity/v2.0`) instead of the default
-value provided by `openrc`. You may also encounter the following issue
-with Cinder:
+Do not use `openshift_hosted_registry_storage_openstack_volumeID` at the same
+as `openshift_hosted_registry_storage_volume_name`.
 
-https://github.com/kubernetes/kubernetes/issues/50461
+The `volumeID`
+will [use an existing Cinder volume's ID for the registry][existing],
+but `volume_name` will actually create a new volume if it doesn't
+exist already.
 
-You can read the (OpenShift documentation on configuring
-OpenStack)[openstack] for more information.
+[existing]: #use-an-existing-cinder-volume-for-the-openshift-registry
 
-[devstack]: https://docs.openstack.org/devstack/latest/
-[openstack]: https://docs.openshift.org/latest/install_config/configuring_openstack.html
-
-
-Next, we need to instruct OpenShift to use the Cinder volume for its
-registry. Again in `OSEv3.yml`:
-
-    #openshift_hosted_registry_storage_kind: openstack
-    #openshift_hosted_registry_storage_access_modes: ['ReadWriteOnce']
-    #openshift_hosted_registry_storage_openstack_filesystem: xfs
-
-The filesystem value here will be used in the initial formatting of
-the volume.
-
-If you're using the dynamic inventory, you must uncomment these two values as
-well:
-
-    #openshift_hosted_registry_storage_openstack_volumeID: "{{ lookup('os_cinder', openshift_openstack_cinder_hosted_registry_name).id }}"
-    #openshift_hosted_registry_storage_volume_size: "{{ openshift_openstack_cinder_hosted_registry_size_gb }}Gi"
+### Validation
 
-But note that they use the `os_cinder` lookup plugin we provide, so you must
-tell Ansible where to find it either in `ansible.cfg` (the one we provide is
-configured properly) or by exporting the
-`ANSIBLE_LOOKUP_PLUGINS=openshift-ansible-contrib/lookup_plugins` environment
-variable.
+1. After the provisioning phase is complete, run `openstack volume list`
+   * A new volume with be created, with the name and size you specified
+   * The `Status` of the volume should be `available` and it should not be
+     attached to anything
+2. After the OpenShift installation finished, run `openstack volume list` again
+   * The `Status` of the volume you created should be `in-use`
+   * The volume should be attached to an infra node
```

Comment:
let's note something for the multi-infra nodes deployments

Reply:
Doesn't the "an infra node" imply this? The cinder volume will be attached to the infra node running the registry. That's it.

If you think that's not clear enough, I'll try to word it better.

Reply:
I meant how should this work, when we have multiple infra nodes provisioned

Reply:
Right, okay. In that case you can only have 1 registry pod running on 1 infra node and the volume will be attached to that node. I'll update the docs.